registry/storage/driver/s3-aws/s3.go

```diff
@@ -714,7 +737,7 @@ func (d *driver) Stat(ctx context.Context, path string) (storagedriver.FileInfo,
 		Bucket: aws.String(d.Bucket),
 		Key:    aws.String(d.s3Path(path)),
 	})
-	if err == nil {
+	if err == nil && headResp.ContentLength != nil {
```

Comment:
Should this be `headResp != nil`?

Reply:
The problem I'm trying to guard against here is `ContentLength` being `nil`; in that case it's better to proceed to the fallback of the `ListObjects` call. I'm less concerned about `headResp` being nil since in theory that should only happen if `err != nil` (but in practice here `headResp` will never be nil).

Reply:
Ah, I see. I was confused by the fact that there's also an `if headResp.ContentLength != nil {}` block inside this block. It would be a small readability improvement to remove the additional conditional, but of course it works as-is.